test/system/030-run.bats

```diff
@@ -753,34 +750,37 @@ EOF
 
 @test "podman run defaultenv" {
     run_podman run --rm $IMAGE printenv
-    is "$output" ".*TERM=xterm" "output matches TERM"
-    is "$output" ".*container=podman" "output matches container=podman"
+    assert "$output" =~ "TERM=xterm" "env includes TERM"
+    assert "$output" =~ "container=podman" "env includes container=podman"
 
     run_podman run --unsetenv=TERM --rm $IMAGE printenv
-    is "$output" ".*container=podman" "output matches container=podman"
-    run grep TERM <<<$output
-    is "$output" "" "unwanted TERM environment variable despite --unsetenv=TERM"
+    assert "$output" =~ "container=podman" "env includes container=podman"
+    assert "$output" != "TERM" "unwanted TERM environment variable despite --unsetenv=TERM"
 
     run_podman run --unsetenv-all --rm $IMAGE /bin/printenv
-    run grep TERM <<<$output
-    is "$output" "" "unwanted TERM environment variable despite --unsetenv-all"
-    run grep container <<<$output
-    is "$output" "" "unwanted container environment variable despite --unsetenv-all"
-    run grep PATH <<<$output
-    is "$output" "" "unwanted PATH environment variable despite --unsetenv-all"
+    for v in TERM container PATH; do
+        assert "$output" !~ "$v" "variable present despite --unsetenv-all"
+    done
 
     run_podman run --unsetenv-all --env TERM=abc --rm $IMAGE /bin/printenv
-    is "$output" ".*TERM=abc" "missing TERM environment variable despite TERM being set on commandline"
+    assert "$output" =~ "TERM=abc" \
+           "missing TERM environment variable despite TERM being set on commandline"
 }
 
 @test "podman run - no /etc/hosts" {
     skip_if_rootless "cannot move /etc/hosts file as a rootless user"
-    tmpfile=$PODMAN_TMPDIR/hosts
-    mv /etc/hosts $tmpfile
```

Comment:
YIKES. This was not only a cross-filesystem move (bad idea), but into a tempfs, so the original hosts file would be lost forever if the test crashed in the wrong place! My replacement is still scary, but a little less so.

Reply:
Can we just not do that please, I run the test locally on my laptop. The test should never ever touch system configuration. This is a very bad idea if this crashes or another application on my system tries to use it while it is moved.

IMO such test should be done in the in container test where we have more control or the environment. Alternatively the test should create a new mount namespace where it would remove this file, however that would be super difficult to do properly.

Reply:
Yes, it is a stupendously bad idea. Thanks for the `$container` suggestion, I will do so tout de suite.